packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -159,6 +159,20 @@ void main() {
     expect(renderBox.size.width, 80.0);
   });
 
+  testWidgets('No destinations', (WidgetTester tester) async {
+    await _pumpNavigationRail(
+      tester,
+      navigationRail: NavigationRail(
+        selectedIndex: null,
+        leading: FloatingActionButton(onPressed: () { }),
+        destinations: const <NavigationRailDestination>[],
+      ),
+    );
+
+    final RenderBox leading = tester.renderObject<RenderBox>(find.byType(FloatingActionButton).at(0));
+    expect(leading.localToGlobal(Offset.zero), const Offset(0, 8.0));
+  });
+
   testWidgets('Renders wider for a destination with a long label - [labelType]=all', (WidgetTester tester) async {
```

Comment:
Other than the checking for spacing with zero destination, could you add more tests to make sure if the rail occupies the same space with 0, 1, and 2 destinations.

Also, if the NavigationRail starts from 0 destinations and is rebuilt with one/two more, we'd better check if each destination appears in the correct place.

Reply:
I'm still a little bit new to the flutter testing framework and not quite sure about how to achieve that, copying and pasting the current test for 1 and 2 destinations cases seems not the correct way to test it. Could someone else help working on the additional tests if required? Thanks!

Reply:
To follow up with what @QuncCccccc mentioned, it would be good to have other tests that test things like a NavitationRail with no items and not leading or trailing widget in both normal and extended forms (to make sure nothing is depending on there being anything in the rail). We could also test the tailing widget with no items and make sure it is in the right place. You can take a look at other tests like these that are already in the test file as a guide.